client/src/socket.js

```diff
@@ -19,7 +23,9 @@ socket.on("connect", () => {
     store.dispatch(removeOfflineUser(id));
   });
   socket.on("new-message", (data) => {
-    store.dispatch(setNewMessage(data.message, data.sender));
+    if (data.recipientId === store.getState().user.id) {
+      store.dispatch(setNewMessage(data.message, data.sender));
+    }
```

Comment:
Unfortunately this isn't the security fix that needed to be addressed, the issue is when a user posts a message to the server, we are making assumptions about the conversationId that gets passed in. 

See if you can add some security in the endpoint where messages get created - we want to make sure that users can only post messages to conversations the have access to 🔒